compiler/rustc_infer/src/infer/combine.rs

```diff
@@ -812,13 +830,19 @@ impl<'tcx> TypeRelation<'tcx> for Generalizer<'_, 'tcx> {
                     }
                 }
             }
+            ty::ConstKind::Placeholder(placeholder) => {
+                if self.for_universe.can_name(placeholder.universe) {
+                    Ok(c)
+                } else {
+                    Err(TypeError::UniverseMismatch {
+                        variable: self.tcx().mk_ty_var(self.for_vid_sub_root).into(),
+                        placeholder: c.into(),
+                    })
+                }
+            }
+            // Need to manually relate as `super_relate_consts` is not simply structural.
             ty::ConstKind::Unevaluated(ty::UnevaluatedConst { def, substs }) => {
```

Comment:
Same here, though this requires changing `super_relate_consts` to not do const evaluation. Probably out of the scope of this PR.